src/devices/vdml_usd.cpp

```diff
@@ -71,29 +71,30 @@ std::vector<std::string> list_files(const char* path, char* buffer, int32_t len)
 		}
 		return files;
 	}
+
 	// Parse buffer given call successful, split by '/n'
-	// Store char * buffer as std::string in str
-	// Store delimiter in std::string, '\n'
-	std::string str(buffer);
-	std::string delimiter = "\n";
+	std::string_view str(buffer);
 
-	// position to keep track of position in str
+	// delimter_pos is the position of the delimiter '\n'
+	// index of which character to start substr from
 	// file_name used to store each file name
-	size_t position = 0;
-	std::string file_name;
+	size_t delimiter_pos, index = 0;
```

Comment:
C++ footgun alert!! `index` is initialized to 0 here, but `delimiter_pos` is left uninitialized (and therefore in an unspecified state). Admittedly, it doesn't matter here since the first use is an assignment, but I generally advise to avoid multiple declaration and always initialize primitive types to avoid even the potential of such mistakes causes issues, since those can be very hard to track down later.